src/Simulation/Simulators/QuantumSimulator/StateDumper.cs

```diff
@@ -60,29 +67,190 @@ public virtual bool Dump(IQArray<Qubit>? qubits = null)
+                bool truncateSmallAmplitudes, double truncationThreshold
+            ) =>
+                (
+                    truncateSmallAmplitudes
+                    ? Amplitudes
+                        .Select((amplitude, idx) => (amplitude, idx))
+                        .Where(item =>
+                            System.Math.Pow(item.amplitude.Magnitude, 2.0) >= truncationThreshold
+                        )
+                    : Amplitudes.Select((amplitude, idx) => (amplitude, idx))
+                )
+                .Select(
+                    item => (item.amplitude, BasisStateLabel(convention, item.idx))
+                )
+                .OrderBy(
+                    item => item.Item2,
+                    // If a basis state label is numeric, we want to compare
+                    // numerically rather than lexographically.
+                    convention switch {
+                        BasisStateLabelingConvention.BigEndian => ToIntComparer,
+                        BasisStateLabelingConvention.LittleEndian => ToIntComparer,
+                        _ => Comparer<string>.Default
+                    }
+                );
+
+            /// <summary>
+            ///     Using the given labeling convention, returns the label for a
+            ///     computational basis state described by its bit string as encoded
+            ///     into an integer index in the little-endian encoding.
+            /// </summary>
+            public string BasisStateLabel(
+                BasisStateLabelingConvention convention, int index
+            ) => convention switch
+                {
+                    BasisStateLabelingConvention.Bitstring =>
+                        String.Concat(
+                            System
+                                .Convert
+                                .ToString(index, 2)
+                                .PadLeft(NQubits, '0')
+                                .Reverse()
+                        ),
+                    BasisStateLabelingConvention.BigEndian =>
+                        System.Convert.ToInt64(
+                            String.Concat(
+                                System.Convert.ToString(index, 2).PadLeft(NQubits, '0').Reverse()
+                            ),
+                            fromBase: 2
+                        )
+                        .ToString(),
+                    BasisStateLabelingConvention.LittleEndian =>
+                        index.ToString(),
+                    _ => throw new ArgumentException($"Invalid basis state labeling convention {convention}.")
+                };
+
 
             /// <summary>
             /// Returns a string that represents the label for the given base state.
             /// </summary>
-            public virtual string FormatBaseState(uint idx) =>
-                $"∣{idx.ToString().PadLeft(_maxCharsStateId, ' ')}❭";
+            public virtual string FormatBaseState(uint idx) 
```

Comment:
FYI: 
The diff of this file does not display the change properly.

The whole `SimpleDumper` with its `Format..()` logic (consisting of 
+ string Format(uint idx, double real, double img)
+ string FormatBaseState(uint idx)
+ string FormatCartesian(double real, double img)
+ string FormatPolar(double magnitude, double angle)
+ string FormatMagnitude(double magnitude, double phase)
+ string FormatAngle(double magnitude, double angle)

) has been removed. 
The `enum BasisStateLabelingConvention` and `class DisplayableState` have been added. The `Format..()` logic has been moved from the removed `SimpleDumper` to the `class DisplayableState`. In that `Format..()` logic the `FormatBaseState()` has been updated slightly. That is why in that `Format..()` logic only the `FormatBaseState()` is shown (by `git diff`) as new, but in reality the whole `Format..()` logic is new in this `class DisplayableState`.

I assume that with your words 
"Since **it**'s public we may not be able to remove it" you mean 
"Since **the `FormatBaseState()`** is public we may not be able to remove it". 
I'd like to admit that the `FormatBaseState()` is totally new for `class DisplayableState`, that is why _we can remove it_.

I was also thinking may be I should encapsulate this `Format..()` logic into a class (probably defined inside of `DisplayableState`) to keep the `Format..()` logic "as a single piece". In which case it would make sense to preserve the `FormatBaseState()`.
Or I can remove the `FormatBaseState()` and replace the call to it with `BasisStateLabel(BasisStateLabelingConvention.LittleEndian, idx)`.

What would you recommend?